### PR TITLE
Remove LCK

### PR DESCRIPTION
**Pull request by szok**

### CODE REVIEW

File: src/ralph_scrooge/models/cost.py [hunk 1]
```diff
@@ -8,9 +8,8 @@
 from django.db import models as db
 from django.utils.translation import ugettext_lazy as _
 
-from lck.django.common.models import WithConcurrentGetOrCreate
-
 from ralph_scrooge.models._tree import MultiPathNode
+from ralph_scrooge.utils.common import WithConcurrentGetOrCreate
```
Comment (mkurek) [inline src/ralph_scrooge/models/cost.py]: imo we don't need `WithConcurrentGetOrCreate` - we'll use built-in `get_or_create`

File: src/ralph_scrooge/models/owner.py [hunk 2]
```diff
@@ -26,7 +25,7 @@ def get_query_set(self):
         )
 
 
-class Owner(TimeTrackable):
+class Owner(db.Model):
```
Comment (mkurek) [inline src/ralph_scrooge/models/owner.py]: what about fields from `TimeTrackable`?

File: src/ralph_scrooge/models/pricing_object.py [hunk 3]
```diff
@@ -112,7 +108,7 @@ def __unicode__(self):
         return '{} - {}'.format(self.manufacturer, self.name)
 
 
-class PricingObject(TimeTrackable, EditorTrackable):
+class PricingObject(db.Model):
```
Comment (mkurek) [inline src/ralph_scrooge/models/pricing_object.py]: what about fields from `TimeTrackable` and `EditorTrackable`?

File: src/ralph_scrooge/models/service.py [hunk 4]
```diff
@@ -112,7 +110,7 @@ class Meta:
         ordering = ['name']
 
 
-class Service(ModelDiffMixin, EditorTrackable, TimeTrackable):
```
Comment (mkurek) [inline src/ralph_scrooge/models/service.py:115]: what about fields from `TimeTrackable` and `EditorTrackable`?

File: src/ralph_scrooge/models/sync.py [hunk 5]
```diff
@@ -7,10 +7,9 @@
 
 from django.db import models as db
 from django.utils.translation import ugettext_lazy as _
-from lck.django.common.models import TimeTrackable
 
 
-class SyncStatus(TimeTrackable):
+class SyncStatus(db.Model):
```
Comment (mkurek) [inline src/ralph_scrooge/models/sync.py]: what about fields from `TimeTrackable`?

File: src/ralph_scrooge/models/team.py [hunk 6]
```diff
@@ -35,7 +31,7 @@ class TeamBillingType(Choices):
     average = _('Average')
 
 
-class Team(TimeTrackable, EditorTrackable, BaseUsage):
+class Team(BaseUsage):
```
Comment (mkurek) [inline src/ralph_scrooge/models/team.py]: what about fields from `TimeTrackable` and `EditorTrackable`?

File: src/ralph_scrooge/models/warehouse.py [hunk 7]
```diff
@@ -7,20 +7,16 @@
...
-    EditorTrackable,
-    Named,
-    TimeTrackable,
-    WithConcurrentGetOrCreate,
-)
 
 
-class Warehouse(TimeTrackable, EditorTrackable, Named,
-                WithConcurrentGetOrCreate):
+class Warehouse(db.Model):
```
Comment (mkurek) [inline src/ralph_scrooge/models/warehouse.py]: what about fields from `TimeTrackable` and `EditorTrackable`?

File: src/ralph_scrooge/utils/cache.py [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+
+import cPickle as pickle
+import sys
+from time import time
+from functools import wraps
+
+
+def memoize(
+    func=None, update_interval=300, max_size=256, skip_first=False,
+    fast_updates=True
```
Comment (mkurek) [inline src/ralph_scrooge/utils/cache.py]: we don't use fast updates  - remove this param

File: src/ralph_scrooge/utils/cache.py [hunk 9]
```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/python
```
Comment (mkurek) [inline src/ralph_scrooge/utils/cache.py:1]: add a comment with link to original source

File: src/ralph_scrooge/utils/common.py [hunk 10]
```diff
@@ -1,145 +1,226 @@
...
+class AttributeDict(dict):
+    """
+    Attribute dict. Used to attribute access to dict
+    """
+    def __init__(self, *args, **kwargs):
+        super(AttributeDict, self).__init__(*args, **kwargs)
+        self.__dict__ = self
+
+
+class WithConcurrentGetOrCreate(object):
```
Comment (mkurek) [inline src/ralph_scrooge/utils/common.py]: no need for this

File: src/ralph_scrooge/models/pricing_object.py (hunk 3 above)
Comment (mkurek) [inline src/ralph_scrooge/models/pricing_object.py]: imo we should keep these 3 models: `TimeTrackable`, `EditorTrackable` and `Named` - copy it from lck to scrooge.

File: src/ralph_scrooge/utils/cache.py (hunk 9 above)
Comment (szok) [reply src/ralph_scrooge/utils/cache.py:1]: done

File: src/ralph_scrooge/utils/models.py [hunk 11]
```diff
@@ -0,0 +1,237 @@
...
+        def __unicode__(self):
+            return self.name
+
+        @property
+        def name_urlencoded(self):
+            """Useful as in {%url some-link argument.name_urlencoded%}."""
+            return urlencode(self.name, safe="")
+
+
+class TimeTrackable(db.Model):
```
Comment (mkurek) [inline src/ralph_scrooge/utils/models.py:60]: add some comment with TODO to use models similar to Ralph's ones after bumping Django: https://github.com/allegro/ralph/blob/ng/src/ralph/lib/mixins/models.py#L34

File: src/ralph_scrooge/utils/models.py [hunk 12]
```diff
@@ -0,0 +1,237 @@
...
+    )
+
+    class Meta:
+        abstract = True
+
+    def __unicode__(self):
+        return self.name
+
+    @property
+    def name_urlencoded(self):
```
Comment (mkurek) [inline src/ralph_scrooge/utils/models.py]: are we using it anywhere?

File: src/ralph_scrooge/utils/models.py [hunk 13]
```diff
@@ -0,0 +1,228 @@
...
+        name = db.CharField(verbose_name=_("name"), max_length=75)
+
+        class Meta:
+            abstract = True
+
+        def __unicode__(self):
+            return self.name
+
+
+# TODO simplify how will django >1.8
```
Comment (xor-xor) [inline src/ralph_scrooge/utils/models.py]: I don't understand this comment - "how will"..?

**Comment by xor-xor** [comment]
:ok:

File: src/ralph_scrooge/utils/models.py [hunk 14]
```diff
@@ -0,0 +1,231 @@
...
+class EditorTrackable(db.Model):
+    """Describes objects authored by users of the application. In the admin,
+    on object creation the ``created_by`` field is set according to the editor.
+    Same goes for modifying an object and the ``modified_by`` field. Works best
+    in integration with TimeTrackable.
+    If you would rather link to your user profile instead of the user object
+    directly, use the ``EDITOR_TRACKABLE_MODEL`` (the same
+    ``"app_name.ModelClass"``syntax as ``AUTH_PROFILE_MODULE``) setting.
+    Note: for automatic editor updates in admin,
+    ``lck.django.common.admin.ModelAdmin`` **MUST** be used instead of the
```
Comment (mkurek) [inline src/ralph_scrooge/utils/models.py]: :scream:

**Comment by mkurek** [comment]
:ok:

**Pull request merged**